docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+As a concrete example, this code uses `findNodeHandle` to tunnel from `MyComponent` through to the `View` rendered by `MyJSComponent`.
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof MyJSComponent>;
+
+  render() {
+    return <MyJSComponent ref={this._captureRef} onSubmit={this._onSubmit} />
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+
+  _onSubmit: () => {
+    const nodeHandle = findNodeHandle(this._ref);
+
+    if (nodeHandle) {
+      UIManager.measure(nodeHandle, () => {});
+    }
+  }
+}
+
+function MyJSComponent(props) {
+  return (
+    <View>
+      <SubmitButton onSubmit={props.onSubmit} />
+    </View>
+  );
+}
+```
+
+We can’t convert this call to `this._ref.measure` because `this._ref` is an instance to `MyJSComponent`, which is not a HostComponent and thus does not have a `measure` function.
+
+`MyJSComponent` renders a View, which is a HostComponent, so we need to get a reference to `View` instead. There are typically two approaches to get what we need. If the component we need to get the ref from is a function component using `forwardRef` is probably the right choice. If it is a class component with other public methods, adding a public method for getting the ref is an option. Here are examples of those two forms:
+
+#### Using `forwardRef`
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof MyJSComponent>;
+
+  render() {
+    return <MyJSComponent ref={this._captureRef} onSubmit={this._onSubmit} />
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+
+  _onSubmit: () => {
+    if (this._ref != null)
+      this._ref.measure(() => {});
+    }
+  }
+}
+
+const MyJSComponent = React.forwardRef((props, forwardedRef) => {
+  return (
+    <View ref={forwardedRef}>
+      <SubmitButton onSubmit={props.onSubmit} />
+    </View>
+  );
+});
+```
+
+#### Using a getter, (note the addition of `getViewRef`)
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof MyJSComponent>;
+
+  render() {
+    return <MyJSComponent ref={this._captureRef} onSubmit={this._onSubmit} />
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+
+  _onSubmit: () => {
+    if (this._ref != null)
+      this._ref.**getViewRef****()**.measure(() => {});
+    }
+  }
+}
+
+class MyJSComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof View>;
+
+  render() {
+    return (
+      <View ref={this._captureRef}>
+        <SubmitButton onSubmit={props.onSubmit} />
+      </View>
+    );
+  }
+
+  getViewRef(): ?React.ElementRef<typeof View> {
+    return this._ref;
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+}
+```
+
+#### Migrating `.measure(`
+
+Let’s take a look at an example calling `UIManager.measure`. This code might look something like this
+
+```javascript
+if (this._scrollView == null || viewRef == null) {
```

Comment:
It is not clear what `this._scrollView` or `viewRef` here are. (Refs? `ref.current` values? if they have the same type, why does only one of them have `Ref` suffix?) It would help to have a slightly more complete example.

Reply:
This was simplified by @sammy-SC here https://github.com/facebook/react-native-website/pull/2935